rendermodules/mesh_lens_correction/MeshAndSolveTransform.py

```diff
@@ -11,616 +11,584 @@
-
-        self.data = data
-        self.indices = indices
-        self.indptr = indptr
-
-        self.A = csr_matrix((data, indices, indptr))
-        self.weights = sparse.eye(weights.size, format='csr')
-        self.weights.data = weights
-        self.create_x0(self.A.shape[1], self.unique_ids.size)
-
-    def MeshAndSolve(self, render,
-                     input_stack, output_stack,
-                     match_collection, sectionId,
-                     nvertex, output_dir, outfile, default_lambda,
-                     translation_factor, lens_lambda, close_stack=True):
-        self.args = {}
-        self.args['render'] = render
-        self.args['input_stack'] = input_stack
-        self.args['output_stack'] = output_stack
-        self.args['match_collection'] = match_collection
-        self.args['sectionId'] = sectionId
-        self.args['nvertex'] = nvertex
-        self.args['output_dir'] = output_dir
-        self.args['out_html_dir'] = output_dir
-        self.args['outfile'] = outfile
-        self.args['regularization'] = {}
-        self.args['regularization']['default_lambda'] = default_lambda
-        self.args['regularization']['translation_factor'] = translation_factor
-        self.args['regularization']['lens_lambda'] = lens_lambda
-        self.args['close_stack'] = close_stack
-
-        # load the matches
-        self.load_matches()
-
-        # load tilespecs
-        self.load_tilespecs()
-
         # condense coordinates
-        self.condense_coords()
+        self.coords = condense_coords(self.matches)
+        self.coords = even_distribution(
+            self.coords,
+            self.tile_width,
+            self.tile_height,
+            10)
+
+        if self.coords.shape[0] == 0:
+            raise MeshLensCorrectionException(
+                    "no point matches left after evening distribution, \
+                     probably some sparse areas of matching")
 
         # create PSLG
-        self.create_PSLG()
+        self.bbox = create_PSLG(
+                self.tile_width,
+                self.tile_height)
 
         # find delaunay with max vertices
-        self.find_delaunay_with_max_vertices()
-
-        self.force_vertices_with_npoints(3)
-        if self.has_hole:
-            self.add_center_point()
-
-        print('returned  %d vertices' % self.mesh.npoints)
-        print('creating A...')
-        self.create_A()
-
-        print('solving...')
-        self.solve()
+        self.mesh, self.area_triangle_par = \
+            find_delaunay_with_max_vertices(
+                self.bbox,
+                self.args['nvertex'])
+
+        # and enforce neighboring matches to vertices
+        self.mesh, self.area_triangle_par = \
+            force_vertices_with_npoints(
+                self.area_triangle_par,
+                self.bbox,
+                self.coords,
+                3)
+
+        try:
+            assert self.mesh.points.shape[0] > 0.5*self.args['nvertex'], \
```

Comment:
Assertions in python are always touchy -- running optimized python (e.g. with the `-O`  flag) disables them altogether which may or may not be what you want at various points here.

Reply:
you're right. I don't want them disabled. How would you suggest I assert or check numerical results? `if` then `raise`?